File: NgWebApp/src/app/features/warehouse/models/batch.ts
export interface Batch {
  id: string;
  key: string;
  productId: string;
  warehouseId: string;
  quantity: number;
  createdAt: string;
  updatedAt: string;
}

export type AddBatch = Pick<
  Batch,
  'key' | 'productId' | 'warehouseId' | 'quantity'
>;

File: BuyIt/WarehouseApi/Batches/Batch.cs
namespace WarehouseApi.Batches;

public class Batch : Contracts.Warehouses.WarehouseBatch
{
    public Guid Id { get; set; }
    public string Key { get; set; } = "";
    public Guid ProductId { get; set; }
    public Guid WarehouseId { get; set; }
    public int Quantity { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}
